app/models/task.py

```diff
@@ -1,5 +1,34 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
+from sqlalchemy import ForeignKey
 from ..db import db
+from datetime import datetime
+from typing import Optional
+
 
 class Task(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    description: Mapped[str]
+    completed_at: Mapped[Optional[datetime]]
+    goal_id: Mapped[Optional[int]] = db.Column(db.Integer, db.ForeignKey("goal.id"))
+    goal: Mapped[Optional["Goal"]] = db.relationship("Goal", back_populates="tasks")
+
+    def to_dict(self):
+        task_dict = {
+            "id": self.id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.completed_at is not None,
```

Comment:
Consider putting this logic in a helper function like `is_complete`.
```suggestion
            "is_complete": task.is_complete()})
```